app/controllers/works_controller.rb

```diff
@@ -0,0 +1,115 @@
+class WorksController < ApplicationController
+
+  def index
+   @works = Work.all
+  end
+
+
+  def show
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+    if @work.save
+      flash[:success] = "#{@work.title.capitalize} Was successfully added!"
+      redirect_to work_path(@work.id)
+      return
+    else
+      flash.now[:error] = "#{@work.title.capitalize} Was NOT successfully added!"
+      render :edit, status: :bad_request
+      return
+    end
+  end
+
+  def edit
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @work = Work.find_by(id: params[:id])
+
+    if @work.nil?
+      head :not_found
+      return
+    elsif @work.update(work_params)
+      flash[:success] = "#{@work.title.capitalize} Was successfully updated!"
+      redirect_to works_path 
+      return
+    else 
+      flash.now[:error] = "#{@work.title.capitalize} Was NOT successfully updated!"
+      render :edit, status: :bad_request 
+      return
+    end
+  end
+
+  def destroy
+    @work = Work.find_by(id: params[:id])
+    if @work.nil?
+      head :not_found
+      return
+    elsif
+      @vote = Vote.where(work_id: @work.id).count > 0
+      @vote = Vote.where(work_id: @work.id).delete_all
```

Comment:
The first assignment is redundant since `@vote` is immediately overwritten.

```suggestion
      @vote = Vote.where(work_id: @work.id).delete_all
```